core/src/main/java/de/danoeh/antennapod/core/service/download/DownloadService.java

```diff
@@ -225,6 +234,11 @@ public void onDestroy() {
             downloadPostFuture.cancel(true);
         }
         unregisterReceiver(cancelDownloadReceiver);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            if (connectionMonitor != null) {
```

Comment:
Is there any way it can be null here? Android services always go through onCreate before going through onDestroy, so I think no null handling is needed (not even in onCreate because that is only called once)